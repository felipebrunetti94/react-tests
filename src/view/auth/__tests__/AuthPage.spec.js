import { fireEvent, render } from "@testing-library/react";
import AuthPage from "../AuthPage";

describe("View :: Auth :: AuthPage", () => {
  describe("quando clica no botão submit", () => {
    it("passa para o onSubmit as info do usuário", () => {
      const onSubmit = jest.fn();
      const authInfo = "auth info";
      const props = {
        title: "test",
        showUsername: true,
        onSubmit,
        isLoading: false,
        error: {},
        updateAuthInfo: () => {},
        authInfo,
      };
      const { getByRole } = render(<AuthPage {...props} />);
      fireEvent.click(getByRole("button", { type: "submit" }));
      expect(onSubmit).toHaveBeenCalledWith(authInfo);
    });
  });

  describe("quando atualiza um campo", () => {
    it("chama o  Update Auth Info com nome e valor do campo atualizado", () => {
      const updateAuthInfo = jest.fn();
      const props = {
        title: "test",
        showUsername: true,
        onSubmit: () => {},
        isLoading: false,
        error: {},
        updateAuthInfo,
        authInfo: {},
      };
      const { getByPlaceholderText } = render(<AuthPage {...props} />);

      fireEvent.change(getByPlaceholderText("Your Name"), {
        target: {
          value: "text",
        },
      });

      expect(updateAuthInfo).toHaveBeenCalledWith({ username: "text" });
    });
  });

  describe("quando receber a prop showUserName", () => {
    it("mostra o campo username", () => {
      const props = {
        title: "test",
        showUsername: true,
        onSubmit: () => {},
        isLoading: false,
        error: {},
        updateAuthInfo: () => {},
        authInfo: {},
      };
      const { getByPlaceholderText } = render(<AuthPage {...props} />);
      expect(getByPlaceholderText("Your Name")).toBeInTheDocument();
    });
  });

  describe("quando estiver em loading", () => {
    it("o botão submit deve estar desabilitado", () => {
      const props = {
        title: "test",
        showUsername: true,
        onSubmit: () => {},
        isLoading: true,
        error: {},
        updateAuthInfo: () => {},
        authInfo: {},
      };
      const { getByRole } = render(<AuthPage {...props} />);

      expect(getByRole("button", { type: "submit" })).toBeDisabled();
    });
  });

  describe("quando recebe a prop error", () => {
    it("deve mostrar o erro", () => {
      const props = {
        title: "test",
        showUsername: true,
        onSubmit: () => {},
        isLoading: false,
        error: { message: "ohno!" },
        updateAuthInfo: () => {},
        authInfo: {},
      };
      const { getByText } = render(<AuthPage {...props} />);

      expect(getByText("ohno!")).toBeInTheDocument();
    });
  });
});
