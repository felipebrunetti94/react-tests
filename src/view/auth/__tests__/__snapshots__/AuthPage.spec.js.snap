// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View :: Auth :: AuthPage matches snapshot 1`] = `
<div
  className="auth-page"
>
  <div
    className="container page"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6 offset-md-3 col-xs-12"
      >
        <h1
          className="text-xs-center"
        >
          test
        </h1>
        <p
          className="text-xs-center"
        >
          <span>
            link
          </span>
        </p>
        <form>
          <fieldset
            className="form-group"
          >
            <input
              className="form-control form-control-lg"
              data-testid="input-username"
              name="username"
              placeholder="Your Name"
              type="text"
              value=""
            />
          </fieldset>
          <fieldset
            className="form-group"
          >
            <input
              className="form-control form-control-lg"
              data-testid="input-email"
              name="email"
              placeholder="Email"
              type="text"
              value=""
            />
          </fieldset>
          <fieldset
            className="form-group"
          >
            <input
              className="form-control form-control-lg"
              data-testid="input-password"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
          </fieldset>
          <button
            className="btn btn-lg btn-primary pull-xs-right"
            type="submit"
          >
            test
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
